discord/threads.py

```diff
@@ -168,16 +168,23 @@ def __str__(self) -> str:
         return self.name
 
     def _from_data(self, data: ThreadPayload):
+        # This data will always exist
         self.id = int(data["id"])
         self.parent_id = int(data["parent_id"])
-        self.owner_id = int(data["owner_id"])
         self.name = data["name"]
         self._type = try_enum(ChannelType, data["type"])
-        self.last_message_id = _get_as_snowflake(data, "last_message_id")
-        self.slowmode_delay = data.get("rate_limit_per_user", 0)
-        self.message_count = data["message_count"]
-        self.member_count = data["member_count"]
-        self.flags: ChannelFlags = ChannelFlags._from_value(data.get("flags", 0))
+
+        # This data may be missing depending on how this object is being created
+        try:
+            self.owner_id = int(data["owner_id"])
+            self.last_message_id = _get_as_snowflake(data, "last_message_id")
+            self.slowmode_delay = data.get("rate_limit_per_user", 0)
+            self.message_count = data["message_count"]
+            self.member_count = data["member_count"]
+            self.flags: ChannelFlags = ChannelFlags._from_value(data.get("flags", 0))
+        except KeyError:
+            pass
```

Comment:
This could cause issues, if one of the earlier keys isn't there then none of the rest will be set even if they do exist.